src/lib/rng/processor_rng/processor_rng.cpp

```diff
@@ -111,6 +130,8 @@ bool Processor_RNG::available()
    {
 #if defined(BOTAN_TARGET_CPU_IS_X86_FAMILY)
    return CPUID::has_rdrand();
+#elif defined(BOTAN_TARGET_ARCH_IS_ARM64) && !defined(BOTAN_TARGET_OS_IS_MACOS) && defined(BOTAN_TARGET_IS_IOS)
```

Comment:
I don't see any reason to here (or elsewhere in the file) guard against macOS/iOS - if CPUID doesn't indicate support, the code will never be invoked, and the only way CPUID can detect support is with getauxval which doesn't exist on Apple.

Reply:
in fact it s more because the assembly does not compile on these.

Reply:
Oh that's unfortunate.

Maybe instead we can use `__builtin_aarch64_rndrrs` / `__builtin_arm_rndrrs` builtins? And then only enable support if the builtins exist.

Reply:
ah good point